flutter/lib/src/binding_wrapper.dart

```diff
@@ -73,37 +73,61 @@ typedef FrameTimingCallback = void Function(
     DateTime startTimestamp, DateTime endTimestamp);
 
 mixin SentryWidgetsBindingMixin on WidgetsBinding {
-  DateTime? _startTimestamp;
-  FrameTimingCallback? _frameTimingCallback;
+  FrameTimingCallback? _onDelayedFrame;
   ClockProvider? _clock;
-
+  Stopwatch? _stopwatch;
+  Duration? _expectedFrameDuration;
+  bool _isTrackingActive = false;
   SentryOptions get _options => Sentry.currentHub.options;
 
   @internal
-  void registerFramesTracking(
-      FrameTimingCallback callback, ClockProvider clock) {
-    _frameTimingCallback ??= callback;
+  void initializeFramesTracking(
+      FrameTimingCallback onDelayedFrame,
+      ClockProvider clock,
+      Duration expectedFrameDuration,
+      Stopwatch stopwatch) {
+    _onDelayedFrame ??= onDelayedFrame;
     _clock ??= clock;
+    _stopwatch ??= stopwatch;
+    _expectedFrameDuration ??= expectedFrameDuration;
   }
 
   @visibleForTesting
   bool isFramesTrackingInitialized() {
-    return _frameTimingCallback != null && _clock != null;
+    return _onDelayedFrame != null &&
+        _clock != null &&
+        _expectedFrameDuration != null &&
+        _stopwatch != null;
+  }
+
+  void resumeTrackingFrames() {
+    _isTrackingActive = true;
+  }
+
+  void pauseTrackingFrames() {
+    // Stopwatch could continue running if we pause tracking in between a frame
+    _stopwatch?.stop();
+    _stopwatch?.reset();
+    _isTrackingActive = false;
   }
 
   @internal
   void removeFramesTracking() {
-    _frameTimingCallback = null;
+    _onDelayedFrame = null;
     _clock = null;
+    _stopwatch = null;
+    _expectedFrameDuration = null;
   }
 
   @override
   void handleBeginFrame(Duration? rawTimeStamp) {
-    try {
-      _startTimestamp = _clock?.call();
-    } catch (_) {
-      if (_options.automatedTestMode) {
-        rethrow;
+    if (_isTrackingActive) {
```

Comment:
We could do early return. It's easier to read the method for two reasons.

- On condition that return early, we can know we can just ignore what comes after.
- It's one layer less of `if` nesting

```dart
if (!_isTrackingActive) {
  return 
}

```

Reply:
I dont think we can early return here because we need to keep call the parent method `super.handleBeginFrame(rawTimeStamp)` afterwards